test/test_c10d.py

```diff
@@ -2069,6 +2069,69 @@ def test(box, unbox):
             unbox=lambda obj: obj["list"][3],
         )
 
+    @skip_if_not_nccl
+    @skip_if_not_multigpu
+    def test_find_unused_parameters_kwarg(self):
+        """
+        Note: this test can be sped up by only running it on a CPU module
+        once DistributedDataParallel supports them.
+        """
+        store = c10d.FileStore(self.file.name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
+
+        class FindUnusedParametersModule(nn.Module):
+            def __init__(self):
+                super(FindUnusedParametersModule, self).__init__()
+                self.fc1 = nn.Linear(2, 10, bias=False)
+                self.fc2 = nn.Linear(10, 4, bias=False)
+                self.fc3 = nn.Linear(4, 4, bias=False)
+                self.relu = nn.ReLU()
+
+            def forward(self, x):
+                x = self.relu(self.fc1(x))
+                x = self.relu(self.fc2(x))
+                # Return the fc3 module so that the caller can invoke it
+                # outside of the forward function. While this is bad practice,
+                # we can use it to trigger a reducer error.
+                return (F.softmax(x, dim=1), self.fc3)
+
+        device_id = gpus_for_rank(self.world_size)[self.rank][0]
+        batch_size = 4
+        criterion = nn.CrossEntropyLoss()
+        input = torch.rand([batch_size, 2], dtype=torch.float)
+        target = torch.LongTensor([random.randrange(4) for _ in range(batch_size)]).to(device_id)
+
+        def test_find_unused_parameters(find_unused_parameters):
+            model = DistributedDataParallel(
+                FindUnusedParametersModule().float().to(device_id),
+                device_ids=[device_id],
+                process_group=process_group,
+                find_unused_parameters=find_unused_parameters,
+            )
+
+            output, fc3 = model(input)
+            output = fc3(output)
+            loss = criterion(output, target)
+            loss.backward()
+
+        # First test that the default behavior under these conditions is to
+        # trigger an error when `backward` is called (because fc3 is an unused
+        # parameter and will therefore be marked ready twice).
+        try:
+            test_find_unused_parameters(True)
+        except Exception as ex:
+            self.assertTrue(
```

Comment:
there is `assertExpectedRaises`

Reply:
Took this approach to make identical to the case below.